tls/s2n_server_cert_verify.c

```diff
@@ -92,14 +104,20 @@ int s2n_server_write_cert_verify_signature(struct s2n_connection *conn, struct s
 
 int s2n_server_generate_unsigned_cert_verify_content(struct s2n_connection *conn, struct s2n_stuffer *unsigned_content, s2n_hash_algorithm chosen_hash_alg)
 {
-    struct s2n_hash_state hash_copy;
+    struct s2n_hash_state selected_hash, hash_copy;
     uint8_t hash_digest_length;
     uint8_t digest_out[S2N_MAX_DIGEST_LEN];
 
+    /* Get current hash content */
+    GUARD(s2n_handshake_get_hash_state(conn, chosen_hash_alg, &selected_hash));
+
     /* Copy current hash content */
-    GUARD(s2n_handshake_get_hash_state(conn, chosen_hash_alg, &hash_copy));
+    GUARD(s2n_hash_new(&hash_copy));
+    GUARD(s2n_hash_copy(&hash_copy, &selected_hash));
+
     GUARD(s2n_hash_digest_size(chosen_hash_alg, &hash_digest_length));
```

Comment:
Do we want to add  the check `lte_check(hash_digest_length, S2N_MAX_DIGEST_LEN);` here?

Reply:
that should not be necessary as S2N_MAX_DIGEST_LEN is the maximum `s2n_hash_digest_size()` would return